README.md

```diff
@@ -41,121 +46,187 @@ You can find a version of this docummentation, properly rendered, at https://doc
 ```
-**Note:** Ensure that you initialize source{d} Engine every time you want to process a new repository. Changes in the `init` working directory are not detected automatically.
 
-**Note for MaOS:** Docker for Mac [requires file sharing](https://docs.docker.com/docker-for-mac/troubleshoot/#volume-mounting-requires-file-sharing-for-any-project-directories-outside-of-users) for any path outside of `/Users`.
+**Note:**
+Ensure that you initialize source{d} Engine every time you want to process a new repository.
+Changes in the `init` working directory are not detected automatically.
+
+**Note for MacOS:**
+Docker for Mac [requires file sharing](https://docs.docker.com/docker-for-mac/troubleshoot/#volume-mounting-requires-file-sharing-for-any-project-directories-outside-of-users) for any path outside of `/Users`.
 
-#### 4. Explore the source{d} Engine
+### 4. Explore the source{d} Engine
+
+The source{d} Engine can [query code repositories](#querying-code) and its capabilities include [parsing code](#parsing-code) into [Universal Abstract Syntax Trees](#babelfish-uast) as well, via different interfaces.
+In this section we will cover a mix of some commands and interfaces available.
+
+**Note:**
+The first time you run some of these commands, the source{d} Engine will download and install the Docker containers as needed. Be aware that this might take a bit of time if it is the first time you use them.
+
+#### Querying code
+
+##### Query web interface
 
 To launch the [web client for the SQL interface](https://github.com/src-d/gitbase-web), run the following command and start executing queries:
 
 ```bash
+# Launch the query web client
 srcd web sql
 ```
 
-The first time you run some of these commands, the source{d} Engine will download and install the Docker containers that are needed. Be aware that this might take a bit of time, it is only on your first use. 
+This should open the [web interface](https://github.com/src-d/gitbase-web) in your browser.
+You can also access it directly at [http://localhost:8080](http://localhost:8080).
 
-If you prefer to stay with the command line, you can execute:
+##### Query command-line interface (CLI)
+
+If you prefer to work within your terminal via command line, you can open a SQL REPL
+that allows you to execute queries against your repositories by executing:
 
 ```bash
+# Launch the query CLI REPL
 srcd sql
 ```
 
-This will open a SQL REPL that allows you to execute queries against your repositories.
-
 If you want to run a query directly, you can also execute it as such:
 
 ```bash
+# Run query via CLI
 srcd sql "SHOW tables;"
 ```
 
-You might have noticed that some queries below use the UAST function. This is to transform code to a [Universal Abstract Syntax Tree](#babelfish-uast). If you want a playground to see examples of the UAST, or run your own, you can launch the parse web client.
+**Note:**
+Queries using the [UAST](#babelfish-uast) function are meant for the `web sql` interface.
```

Comment:
```suggestion
Engine's SQL supports a [UAST](#babelfish-uast) function that returns a Universal AST for the selected source text. UAST values are returned as binary blobs, and are best visualized in the `web sql` interface rather than the CLI.
```